collated/main/A0135768R.md

```diff
@@ -0,0 +1,1803 @@
+
+    public static Optional<LocalDateTime> parseStringIntoDateTime (String rawString) {
+
+        if (rawString == null) {
+            return Optional.empty() ;
+        }
+        
+	    if (rawString.equals(STRING_FOR_DELETE)) {	
+	    	return Optional.of(MARKER_FOR_DELETE);
+	    }
+	    
+
+        Parser dateParser = new Parser() ;
+
+        List<DateGroup> dates = dateParser.parse(rawString) ;
+
+        try {
+            Date date = dates.get(0).getDates().get(0) ;
+
+            return Optional.of(LocalDateTime.ofInstant(date.toInstant(), ZoneId.systemDefault())) ;
+
+        } catch (IndexOutOfBoundsException e) {
+            return Optional.empty() ;
+        }
+    }
+
+    public static String parseLocalDateTimeIntoString (LocalDateTime datetime) {
+        assert datetime != null ;
+
+        return datetime.format(FORMATTER) ;
+    }
+
+    /**
+     * Given a new event that is being added/to be added to the model, checks that if any events currently in the model
+     * that will conflict with the given event. If the given event is already added to the model, it will be ignored - that is
+     * an event cannot conflict with itself.
+     * 
+     * @param model     The model of the TaskForce - cannot be null.
+     * @param eventToBeAdded    The event that is being added/or already added to the model.
+     * @return      {@link Optional.empty()} if there is no conflict; otherwise an arbitrary conflicting event is returned.
+     */
+    public static Optional<Event> checkForConflictingEvents (Model model, Event eventToBeAdded) {
+        assert model != null ;
+
+        LocalDateTime startTime = eventToBeAdded.getStartDate() ;
+        LocalDateTime endTime = eventToBeAdded.getEndDate() ;
+
+        assert startTime.isBefore(endTime) ;
+
+        Long days = ChronoUnit.DAYS.between(startTime, endTime) ;
+
+        model.searchTaskList(new PredicateExpression(new NameQualifier(Sets.newHashSet(days.toString()), NameQualifier.FILTER_BY_DAY, false)) );
+
+        PriorityQueue<Event> pq = new PriorityQueue<>(new EventComparator()) ;
+
+        for (ReadOnlyTask task : model.getSearchedTaskList()) {
+
+            if ( !(task instanceof Event) || task.equals(eventToBeAdded) ) {
+                continue ;
+            }
+
+            pq.add((Event) task) ;
+
+        }
+
+        // Don't need to do anything if there is no events occuring on this time period
+        if (pq.isEmpty()) {
+            return Optional.empty() ;
+        }
+
+        Event event = pq.poll() ;
+
+        while (event.getEndDate().isAfter(startTime)) {
+
+            if (event.getStartDate().isBefore(endTime)) {
+                return Optional.of(event);
+            }
+
+            event = pq.poll() ;
+        }
+
+        return Optional.empty() ;
+
+    }
+
+
+    private static Optional<Pair<LocalDateTime, LocalDateTime>> determineStartAndEndDateTime(Optional<LocalDateTime> startDate, Optional<LocalDateTime> endDate, boolean canBeEmpty) {
```

Comment:
The method parameter names should carefully named. 
"canBeEmty" this is wrong. Usually when you name a boolean variable: isEmpty, isSomething format is use as a standard. I prefer if you cahnge this.